app/src/main/java/com/vultisig/wallet/ui/models/swap/SwapFormViewModel.kt

```diff
@@ -155,26 +156,31 @@ internal class SwapFormViewModel @Inject constructor(
             val gasFeeFiatValue = gasFeeFiat.value ?: return
 
             val srcToken = selectedSrc.account.token
-            val dstToken = selectedDst.account.token
-
-            if (srcToken == dstToken) {
-                throw InvalidTransactionDataException(
-                    UiText.StringResource(R.string.swap_screen_same_asset_error_message)
-                )
-            }
-
-            val srcAddress = selectedSrc.address.address
+            val selectedSrcBalance = selectedSrc.account.tokenValue?.value ?: return
 
             val srcAmountInt = srcAmount
                 ?.movePointRight(selectedSrc.account.token.decimal)
                 ?.toBigInteger()
 
-            val selectedSrcBalance = selectedSrc.account.tokenValue?.value ?: return
             if (srcAmountInt == BigInteger.ZERO) return
+
             val srcTokenValue = srcAmountInt
                 ?.let { convertTokenAndValueToTokenValue(srcToken, it) }
                 ?: return
 
+            val quote = quote ?: return
+
+            showLoading()
+
+            val dstToken = selectedDst.account.token
+
+            if (srcToken == dstToken) {
+                throw InvalidTransactionDataException(
+                    UiText.StringResource(R.string.swap_screen_same_asset_error_message)
+                )
+            }
+
+            val srcAddress = selectedSrc.address.address
```

Comment:
Because we  the `return` statements is preventing from executing `hideLoading() ` and that's  why  I Put the `return' statement toghter to prvent the case